### PR TITLE
[thisoldhouse] Add metadata parsing

### PR DESCRIPTION
**Pull request by Jessecar96**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
This PR adds metadata for This Old House episodes such as season, episode, and series.
Most videos on this site are behind a paywall. I've used my paid account to create this.
Added a new test for this and verified that it passes as well.

### CODE REVIEW

File: youtube_dl/extractor/thisoldhouse.py [hunk 1]
```diff
@@ -41,4 +47,24 @@ def _real_extract(self, url):
             video_id = try_get(
                 drupal_settings, lambda x: x['jwplatform']['video_id'],
                 compat_str) or list(drupal_settings['comScore'])[0]
-        return self.url_result('jwplatform:' + video_id, 'JWPlatform', video_id)
+
+        series = self._search_regex(
+            r'episode-breadcrumb(?:.*?)>(?:.*?)>(.*?)<\/a>', webpage,
+            'series name', fatal=False, flags=re.DOTALL).replace(' TV', '')
```
Comment (dstftw) [inline youtube_dl/extractor/thisoldhouse.py]: Breaks on None.

File: youtube_dl/extractor/thisoldhouse.py [hunk 2]
```diff
@@ -41,4 +47,24 @@ def _real_extract(self, url):
...
+
+        season_number = int_or_none(self._search_regex(
+            r'Season (\d+);', webpage, 'season number',
+            fatal=False))
+        episode_number = int_or_none(self._search_regex(
+            r'Season (?:\d+);(?:[\s\S]*)Ep\.(\d+)', webpage, 'episode number',
```
Comment (dstftw) [inline youtube_dl/extractor/thisoldhouse.py]: First two groups are useless.

File: youtube_dl/extractor/thisoldhouse.py [hunk 3]
```diff
@@ -1,24 +1,30 @@
...
-            'id': '2REGtUDQ',
+            'id': '8WrwQuEr',
             'ext': 'mp4',
-            'title': 'How to Build a Storage Bench',
-            'description': 'In the workshop, Tom Silva and Kevin O\'Connor build a storage bench for an entryway.',
-            'timestamp': 1442548800,
-            'upload_date': '20150918',
-        }
+            'title': 'Taking Modern Back to the Future | Brookline Mid-Century Modern House',
+            'description': 'After months of hard work, the lackluster mid-century box is a modern marvel once again. Kevin, Tommy and Richard tour the home and review all the special features that went into this beautiful space Sunil and Neha can now call home.',
```
Comment (dstftw) [inline youtube_dl/extractor/thisoldhouse.py]: `md5:`.

File: youtube_dl/extractor/thisoldhouse.py [hunk 4]
```diff
@@ -1,24 +1,30 @@
...
+from ..utils import (
+    try_get,
+    int_or_none
+)
 
 
 class ThisOldHouseIE(InfoExtractor):
     _VALID_URL = r'https?://(?:www\.)?thisoldhouse\.com/(?:watch|how-to|tv-episode)/(?P<id>[^/?#]+)'
     _TESTS = [{
-        'url': 'https://www.thisoldhouse.com/how-to/how-to-build-storage-bench',
```
Comment (dstftw) [inline youtube_dl/extractor/thisoldhouse.py]: Don't touch the old test.

File: youtube_dl/extractor/thisoldhouse.py (hunk 1 above)
Comment (Jessecar96) [reply youtube_dl/extractor/thisoldhouse.py]: Fixed, videos without series/episode info will no longer break.

File: youtube_dl/extractor/thisoldhouse.py (hunk 3 above)
Comment (Jessecar96) [reply youtube_dl/extractor/thisoldhouse.py]: Added

File: youtube_dl/extractor/thisoldhouse.py (hunk 4 above)
Comment (Jessecar96) [reply youtube_dl/extractor/thisoldhouse.py]: Added the old test back, and this one as a 2nd.

File: youtube_dl/extractor/thisoldhouse.py (hunk 2 above)
Comment (Jessecar96) [reply youtube_dl/extractor/thisoldhouse.py]: Fixed

**Comment by Jessecar96** [comment]
I have made the changes you requested.

File: youtube_dl/extractor/thisoldhouse.py [hunk 5]
```diff
@@ -41,4 +59,26 @@ def _real_extract(self, url):
             video_id = try_get(
                 drupal_settings, lambda x: x['jwplatform']['video_id'],
                 compat_str) or list(drupal_settings['comScore'])[0]
-        return self.url_result('jwplatform:' + video_id, 'JWPlatform', video_id)
+
+        series = self._search_regex(
+            r'episode-breadcrumb(?:.*?)>(?:.*?)>(.*?)<\/a>', webpage,
```
Comment (dstftw) [inline youtube_dl/extractor/thisoldhouse.py]: Superflous groups. Do not capture empty string.

File: youtube_dl/extractor/thisoldhouse.py [hunk 6]
```diff
@@ -41,4 +59,26 @@ def _real_extract(self, url):
...
+            'series name', default=None, flags=re.DOTALL)
```
Comment (dstftw) [inline youtube_dl/extractor/thisoldhouse.py]: Move flags into regex.

File: youtube_dl/extractor/thisoldhouse.py (hunk 5 above)
Comment (dstftw) [inline youtube_dl/extractor/thisoldhouse.py]: No escapes for slash.

**Comment by Jessecar96** [comment]
- Moved flags into regex
- Removed empty capture groups
- Got rid of the escape for slashes

File: youtube_dl/extractor/thisoldhouse.py (hunk 3 above)
Comment (dstftw) [reply youtube_dl/extractor/thisoldhouse.py]: Nothing changed.

File: youtube_dl/extractor/thisoldhouse.py [hunk 7]
```diff
@@ -1,9 +1,14 @@
...
 from __future__ import unicode_literals
 
+import re
+
 from .common import InfoExtractor
 from ..compat import compat_str
-from ..utils import try_get
+from ..utils import (
+    try_get,
+    int_or_none
```
Comment (dstftw) [inline youtube_dl/extractor/thisoldhouse.py]: Alphabetic.

File: youtube_dl/extractor/thisoldhouse.py [hunk 8]
```diff
@@ -41,4 +59,26 @@ def _real_extract(self, url):
             video_id = try_get(
                 drupal_settings, lambda x: x['jwplatform']['video_id'],
                 compat_str) or list(drupal_settings['comScore'])[0]
-        return self.url_result('jwplatform:' + video_id, 'JWPlatform', video_id)
+
+        series = self._search_regex(
+            r'(?s)episode-breadcrumb.*?>.*?>(.*?)</a>', webpage,
```
Comment (dstftw) [inline youtube_dl/extractor/thisoldhouse.py]: >Do not capture empty string.

**Pull request closed without merging**